src/mcts/search.cc

```diff
@@ -902,10 +905,15 @@ void SearchWorker::DoBackupUpdate() {
         full_depth_updated = n->UpdateFullDepth(&cur_full_depth);
       // Best move.
       if (n->GetParent() == search_->root_node_) {
-        if (!search_->best_move_edge_ ||
-            search_->best_move_edge_.GetN() < n->GetN()) {
+        auto best_n = search_->best_move_edge_.GetN();
+        auto this_n = n->GetN();
+        if (!search_->best_move_edge_ || best_n < this_n) {
```

Comment:
This logic was written this way originally I suspect because best_move_node was a pointer, now that best_move_edge_ is a struct its GetN will be 0, and so this logic can be simplified?

Reply:
Yes that's what I wondered in the last sentence of my commit message, now that it's initialized by the constructor, it should never be null-equivalent any more. But there are several such checks around the code, and like I said, I'm not willing to claim that removing those checks won't introduce further bugs somehow, not at the moment at least. (That is to say, I think it's more likely to introduce bugs than my currently waiting depth PR.)

What we *really* ought to do is get a proper test suite operational, like I opined about way back when on Discord (around the time of the overfitting debacle I think). Unfortunately, for now, test pipe promo takes precedence.

Reply:
My comment was actually that even if it is null equivalent, GetN() returns 0, so at worst it will fall in to the GetBestChildNoTemperature path.

Reply:
Ah, now I understand. I'll think about it (eventually).